### ISSUE TITLE
Command output view

### ISSUE THREAD

**Issue by prosoitos** [ranger/ranger#1599]
When I run commands from ranger, sometimes the output pops up, sometimes it does not. Once I quit ranger, I can see all the outputs of that session in the terminal.

Is there a way to see this output, which accumulates in the terminal, without having to close ranger? Often, I have to quit ranger and relaunch it just to see whether a command completed successfully or not, whether there is an error message, etc., etc.

Of course, I can redirect the output to the pager if I want to make sure that I don't miss any of the outputs. But the pager does always show (even when there is no output), so that makes this solution quite unpractical (by the way, I like the 2nd suggestion of #884: not showing the pager when there is no output would allow to run `shell -p` more indiscriminately).

There might be something already that I missed despite looking for it a fair bit (apologies if that is the case). But if not, would it be possible to implement the equivalent of `display_log` or `taskview_open` but with all the terminal outputs of one ranger session?

**Comment by toonn** [comment]
I usually just do a quick `S<c-d>`. I like all of #884, just not enough time in a day unfortunately.

**Comment by prosoitos** [comment]
`S<c-d>`??? What is this bound to? I can't find what that does in the rc.conf (and I have re-mapped so many keys that I can't test it on my setup).

> I like all of #884, just not enough time in a day unfortunately.

Fair enough! Thank you very much for all the work and the amazing program!

**Comment by prosoitos** [comment]
> I usually just do a quick S<c-d>

Oh, I got it.

**Comment by prosoitos** [comment]
I see. It wasn't working for me because I had replaced `shell $SHELL` by `shell -t $HOME/bin/shellcd %d`, shellcd being a script to change the shell path to the currently visited ranger path.

I guess I will bind both to different keys. `shell -t $HOME/bin/shellcd %d` is better (I think) to run things, but `shell $SHELL` is still necessary to be able to see the output of the shell commands run directly from ranger.

Thanks for the suggestion.

**Comment by toonn** [comment]
Wait, what? `S` already drops me in a shell in the current ranger directory.

**Comment by prosoitos** [comment]
Does it? Do you have something in your settings that may do this? It doesn't for me.

**Comment by prosoitos** [comment]
I run an alias for ranger which makes my shell move to the last visited ranger directory when I quit ranger. Maybe that is messing it up for me? I'll have to test things.

**Comment by toonn** [comment]
Nope, nothing. It's just what happens, for anyone I know. You must have misconfigured either ranger or your shell.

You can test the default keybindings by running `ranger --clean` btw. That keeps ranger from loading your personal config.

**Comment by prosoitos** [comment]
ok. Thank you for confirming. I am also running an alias which allows me to have access to zsh aliases from within ranger. So one of those aliases must be creating this for me.

**Comment by prosoitos** [comment]
This is where I found the code I am using (I am on Arch Linux): https://wiki.archlinux.org/index.php/Ranger#Start_a_shell_from_ranger

I just noticed in the ranger wiki a note mentioning that the arch wiki contains outdated information. So this may be part of what is outdated, but for some reason it is still valid for me.

**Comment by prosoitos** [comment]
And I removed all my aliases and I have the same behaviour.

**Comment by toonn** [comment]
Don't know what's going on. I'm on arch too and it has never not worked for me.
